src/main/java/seedu/cakecollate/model/order/ContainsKeywordsPredicate.java

```diff
@@ -0,0 +1,119 @@
+package seedu.cakecollate.model.order;
+
+import static seedu.cakecollate.logic.parser.CliSyntax.PREFIX_ADDRESS;
+import static seedu.cakecollate.logic.parser.CliSyntax.PREFIX_DATE;
+import static seedu.cakecollate.logic.parser.CliSyntax.PREFIX_DELIVERY_STATUS;
+import static seedu.cakecollate.logic.parser.CliSyntax.PREFIX_EMAIL;
+import static seedu.cakecollate.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.cakecollate.logic.parser.CliSyntax.PREFIX_ORDER_DESCRIPTION;
+import static seedu.cakecollate.logic.parser.CliSyntax.PREFIX_PHONE;
+import static seedu.cakecollate.logic.parser.CliSyntax.PREFIX_REQUEST;
+import static seedu.cakecollate.logic.parser.CliSyntax.PREFIX_TAG;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+
+import seedu.cakecollate.commons.util.StringUtil;
+import seedu.cakecollate.logic.parser.Prefix;
+import seedu.cakecollate.model.tag.Tag;
+
+/**
+ * Tests that an {@code Order}'s {@code Name} matches any of the keywords given.
+ */
+public class ContainsKeywordsPredicate implements Predicate<Order> {
+    private final HashMap<Prefix, List<String>> keywords;
+
+    public ContainsKeywordsPredicate(HashMap<Prefix, List<String>> keywords) {
+        this.keywords = keywords;
+    }
+
+    @Override
+    public boolean test(Order order) {
```

Comment:
Seems like this method takes care of the "And" search, I'm not too sure about which part of the code takes care of the "or" search when no prefixes are specified?

Reply:
"or" search is done by the "all/" prefix which concatenate all the fields in an order to search the keywords against. So that becomes like alloforder.contains(keywords)